### ISSUE TITLE
Lower `pub` verbosity in `flutter -v` mode

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#121001]
Right now when `flutter` is running in verbose mode, we make `pub` run in verbose mode too. Unfortunately `pub`'s verbose mode is really _really_ verbose and it's drowning out everything else in the logs.

**Comment by christopherfujino** [comment]
@sigurdm if we use summary only mode for pub when running flutter verbosely, will we still see the relevant information if there's some kind of http error?

**Comment by sigurdm** [comment]
It makes sense to not run pub in verbose mode when it is run as a prestep of eg. `flutter run`. It is indeed very verbose, and not necessarily what you are interested in.

> @sigurdm if we use summary only mode for pub when running flutter verbosely, will we still see the relevant information if there's some kind of http error?

In summary mode, if resolution fails (eg. because of a http error) you will get a line like:

```
Resolving dependencies failed. For details run `flutter pub get`
```

**Comment by christopherfujino** [comment]
> It makes sense to **not** run pub in verbose mode when it is run as a prestep of eg. `flutter run`. It is indeed very verbose, and not necessarily what you are interested in.

We would be interested if pub failed due to an HTTP error on CI, which it sounds like summary mode would not be sufficient for.

**Comment by sigurdm** [comment]
In that case we probably have to invent a new verbosity-level or tweak some things.
Perhaps we can adjust the summary-only mode such that for failures you get a full (normal verbosity) description of the error, but still only a show a single line for the success case...

We could also consider dumping the logs to a file in case of error, and pointing to that file - but that might not be desirable in CI.

cc @jonasfj who might have good ideas.

**Comment by christopherfujino** [comment]
Yeah, I would prefer summary get tweaked to include more details on an error. Otherwise, I would probably have to update `flutter -v` to use `Process.run(['dart', 'pub', 'get', '-v'])` and then only conditionally print the output if the exit was non-zero.

**Comment by sigurdm** [comment]
> Otherwise, I would probably have to update flutter -v to use Process.run(['dart', 'pub', 'get', '-v']) and then only conditionally print the output if the exit was non-zero.

That sounds super-undesirable! ;)

I think I have convinced @jonasfj to let summary-mode give the regular error description in case resolution fails. That would still not be as verbose as --verbose, but correspond to the normal output of `flutter pub get`.

How does that sound?

**Comment by christopherfujino** [comment]
> > Otherwise, I would probably have to update flutter -v to use Process.run(['dart', 'pub', 'get', '-v']) and then only conditionally print the output if the exit was non-zero.
> 
> That sounds super-undesirable! ;)
> 
> I think I have convinced @jonasfj to let summary-mode give the regular error description in case resolution fails. That would still not be as verbose as --verbose, but correspond to the normal output of `flutter pub get`.
> 
> How does that sound?

This sounds ideal :)

**Comment by eliasyishak** [comment]
@sigurdm  Am I correct to assume this issue has been fixed in dart-lang/pub#3804?  If so, feel free to close this issue
